pocs/utils/logger.py

```diff
@@ -18,14 +45,18 @@ def get_root_logger(profile='panoptes', log_config=None):
     # Get log info from config
     log_config = log_config if log_config else load_config('log').get('logger', {})
 
+    log_dir = '{}/logs'.format(os.getenv('PANDIR', '/var/panoptes/'))
+    log_fname = 'panoptes-{}-{}.log'.format(os.getpid(), datetime.datetime.utcnow().strftime('%Y%m%dT%H%M%SZ'))
+    log_fname_generic = 'panoptes.log'
+
     # Alter the log_config to use UTC times
     if log_config.get('use_utc', True):
         for name, formatter in log_config['formatters'].items():
             log_config['formatters'][name].setdefault('()', _UTCFormatter)
 
     log_file_lookup = {
-        'all': "{}/logs/panoptes.log".format(os.getenv('PANDIR', '/var/panoptes')),
-        'warn': "{}/logs/warnings.log".format(os.getenv('PANDIR', '/var/panoptes')),
+        'all': "{}/{}".format(log_dir, log_fname),
+        'warn': "{}/warnings.log".format(log_dir),
```

Comment:
I'm unclear on how this lookup is used.

I'm used to a model where we do actually have 3 sets of log files for each execution, one for each level INFO, WARN and ERROR (actually, there is also a FATAL file too, created when a program is crashing). Each file contains all messages of its level, and all the more severe levels. And each file has the same name except for the text INFO, WARN (or maybe WARNING) or ERROR.

Reply:
I'm not opposed to this or to naming the file after the invoked script but we now have to have some kind of log file maintenance in place beyond the simple rotation we had before. I already have 20+ log files in the directory that are generated during development.

Reply:
Sorry, but I laughed at that. I periodically clean up the thousands of log files in my ~/tmp directory at work.

Reply:
Yeah, it will only really be a problem for developers. A running unit shouldn't be switching all that much.

Regardless, I created #146